projects/UnitTest/CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+# CMakeList.txt : CMake project for XbSymbolDatabaseUnitTest
+#
+cmake_minimum_required(VERSION 3.11)
+
+project(XbSymbolDatabaseUnitTest LANGUAGES CXX)
+
+# Enforce warnings as errors to verify projects are in sync and doesn't have hidden warnings.
+if(${CMAKE_VERSION} VERSION_LESS "3.24.0")
```

Comment:
Should probably drop support to make this a lot simpler.
Worst case is that users on ancient machines will be able to run, even with warnings. You'd still catch this on CI or when testing a contribution.